prov/efa/test/efa_unit_test_common.c

```diff
@@ -154,14 +154,14 @@ void efa_unit_test_resource_destruct(struct efa_resource *resource)
 		assert_int_equal(fi_close(&resource->ep->fid), 0);
 	}
 
-	if (resource->eq) {
```

Comment:
In order to get the EFA UT's passing I needed to switch CQ/EQ destruction ordering b/c I was hitting an assert in [ofi_monitors_cleanup();](https://github.com/ofiwg/libfabric/blob/main/src/fabric.c#L949) -> [monitor_list[i]->cleanup(monitor_list[i]);](https://github.com/ofiwg/libfabric/blob/main/prov/util/src/util_mem_monitor.c#L331) -> [assert(dlist_empty(&monitor->list));](https://github.com/ofiwg/libfabric/blob/main/prov/util/src/util_mem_monitor.c#L171).  Still investigating this.

Reply:
This is happening b/c when we go into [ofi_eq_cleanup()](https://github.com/ofiwg/libfabric/blob/main/prov/util/src/util_eq.c#L233-L234), we are exiting early b/c of a reference counter:
```
	if (ofi_atomic_get32(&eq->ref))
		return -FI_EBUSY;
```
b/c endpoint deletion doesn't actually complete until we call `fi_cq_read()` again, or we call close the CQ.  I will search man pages to see if this ordering is defined.  If it is not defined, this PR will have issues with breaking backwards compatibility.

Reply:
The [docs](https://ofiwg.github.io/libfabric/main/man/fi_eq.3.html) say: `The EQ must not be bound to any other objects prior to being closed, otherwise the call will return -FI_EBUSY.` 

This change moves EP resource freeing to fi_close(CQ). The user can call fi_close(EP), and then fi_close(EQ) and this will now fail (as docs say it will) b/c the implementation change.  This will be confusing for users b/c from their perspective, they have only bound the EP to the EQ, and the EQ, and CQ should be unrelated, so there should be no reason why they need to delete one before the other.

This change "messes" with backwards compatibility for apps who choose to delete other domain resources (such as EQ) before deleting the CQ.  For an app that is compiled with debug libfabric, they will hit assertions.  For an app that is using a production libfabric, they will get error codes returned from `fi_close()`.  The apps can either try to free resources in a different order, or exit without all libfabric resources properly cleaned up.  This is not great behavior but it is better than a segfault.